Automattic/jetpack
# Boost: Add Cloud CSS showstopper error

Fixes 38-gh-Automattic/boost-shield-api

#### Changes proposed in this Pull Request:

* Cloud CSS status store is refactored to align with Critical CSS status store
* Display errors on the following conditions:
  * Cloud CSS generation request to WordPress.com API is timed out.
  * Cloud CSS is not returned after a defined time (Currently set to one hour).
  * Cloud CSS generation fails with a showstopper error.

#### Jetpack product discussion


NA

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Define constant `JETPACK_BOOST_CLOUD_CSS` as true to enable Cloud Critical CSS.
* Navigate to Jetpack boost dashboard: `wp-admin/admin.php?page=jetpack-boost`
* Enable `Automatically Optimize CSS Loading`
* Confirm that Critical CSS is generated.
* Enable Jetpack Sandbox with patch `D78168`
* Simulate a request timeout or error to WordPress.com API sandbox
* Regenerate Cloud CSS by clicking the `regenerate` button
* Confirm that the appropriate error is displayed under the `Automatically Optimize CSS Loading` meta section.
* Simulate Cloud CSS never returning via `WPCOM_REST_API_V2_Endpoint_Jetpack_Boost_Cloud_CSS::update_cloud_css`
* Regenerate Cloud CSS by clicking the `regenerate` button
* Confirm that the timeout error is displayed under the `Automatically Optimize CSS Loading` meta section.
* Simulate a showstopper error
* Regenerate Cloud CSS by clicking the `regenerate` button
* Confirm that the appropriate error is displayed under the `Automatically Optimize CSS Loading` meta section.
* Disable Cloud CSS by setting `JETPACK_BOOST_CLOUD_CSS` as `false`.
* Ensure that local Critical CSS works as expected

Note: This PR needs to be tested in conjunction with D78168-code and latest Hydra
